src/api.js

```diff
@@ -65,13 +69,138 @@ const requestPcsSummary = ( lang, title, callback, request = cachedRequest ) =>
 
 }
 
+const simplify = ( node ) => {
+	if ( !node ) {
+		return null
+	}
+	// remove a bunch of things
+	const selector = [
+		'script',
+		'meta',
+		'style',
+		'figure',
+		'table',
+		'sup.mw-ref',
+		'sup.reference',
+		'.pcs-collapse-table-container',
+		'.thumb',
+		'.hatnote',
+		"[ role='navigation' ]",
+		'#pcs-edit-section-add-title-description'
+	].join( ',' )
+	for ( const n of node.querySelectorAll( selector ) ) {
+		n.remove()
+	}
+
+	// unwrap links
+	for ( const a of node.querySelectorAll( 'a' ) ) {
+		a.outerHTML = a.innerHTML
+	}
+
+	// remove phonetic notations
+	for ( const p of node.querySelectorAll( 'p' ) ) {
+		p.innerHTML = p.innerHTML.replace( /\s\(.*?class=".*?(ext-phonos|IPA).*?".*?\)/g, '' )
+	}
+
+	if ( node.innerText.trim() === '' ) {
+		return null
+	}
+
+	return node.outerHTML
+}
+
+const findLeadImageThumbnail = ( doc ) => {
+	const leadImageElement = doc.querySelector( 'meta[property="mw:leadImage"]' )
+	if ( !leadImageElement ) {
+		return null
+	}
+	const parts = leadImageElement.getAttribute( 'content' ).split( '/' )
+	const filename = decodeURIComponent( parts[ parts.length - 1 ] )
+	const a = doc.querySelector( 'a[href*="' + filename + '"]' )
+	if ( !a ) {
+		return null
+	}
+	const span = a.querySelector( 'span[data-src]' )
+	if ( span ) {
+		return span.getAttribute( 'data-src' )
+	}
+	const img = a.querySelector( 'img[src]' )
+	if ( img ) {
+		return img.getAttribute( 'src' )
+	}
+	return null
+
+}
+
+const extractSectionSummary = ( lang, title, sectionId, callback, request ) => {
+	const url = `https://${ lang }.wikipedia.org/api/rest_v1/page/mobile-html/${ encodeURIComponent( title ) }?${ getAnalyticsQueryParam() }`
+	request( url, ( data, err ) => {
+		if ( !data ) {
+			logError( msg( lang, 'preview-console-error-message', title, lang ), err )
+			return false
+		}
+		const doc = new DOMParser().parseFromString( data, 'text/html' )
+		const leadImageUrl = findLeadImageThumbnail( doc )
+		const sections = Array.from( doc.querySelectorAll( 'section' ) ).map( ( sectionElement ) => {
+			const sectionTitleElement = sectionElement.querySelector( 'h2, h3, h4, h5, h6' )
+			if ( !sectionTitleElement ) {
+				return null
+			}
+
+			const imageElement = sectionElement.querySelector( 'figure span.mw-file-element' )
+			const imgUrl = imageElement ?
+				imageElement.getAttribute( 'data-src' ) :
+				leadImageUrl
+
+			const extractHtml = simplify(
+				sectionElement.querySelector( 'p' )
+			)
+
+			return extractHtml ? {
+				id: sectionTitleElement.id,
+				imgUrl,
+				extractHtml
+			} : null
+		} ).filter( ( s ) => s )
+		return {
+			sections,
+			dir: doc.body.getAttribute( 'dir' )
+		}
+	}, ( info ) => {
+		for ( const section of info.sections ) {
+			if ( section.id === sectionId ) {
+				callback( {
+					title: title + '#' + sectionId,
+					extractHtml: section.extractHtml,
+					imgUrl: section.imgUrl,
+					dir: info.dir,
+					type: 'standard'
+				} )
+			}
```

Comment:
question, the callback doesn't break here?

Reply:
Yes good catch, it should return/break after the matching section callback, otherwise we always end up with `No preview available for this section` in the Wordpress plugin selector